### DISCUSSION TITLE
當好幾個詞庫都有詞頻時，我應該怎麼調整可以達到比較好的效果？

### DISCUSSION THREAD

**Discussion by markwu** [rime/home#1737] [General]
請問各位：

最近我開始轉換自己的字典來使用。

包含酷音輸入法、小麥輸入法等有詞頻的檔案。另外也轉換了其他無詞頻的資料，如道路、村里和地區資料庫來讓輸入法更正確。

沒有詞頻的資料相對簡單，轉換後我只要使用 `use_preset_vocabulary: true` 讓他去參照八股文中的詞頻即可。

那麼這些有詞頻的字典檔呢？

我應該把  `use_preset_vocabulary: true`  拿掉？還是要保留？

針對字典檔本身，我應該做什麼樣的調整會讓他的輸入效果比較好呢？

保留詞頻，和八股文是否會有衝突呢？

**Comment by markwu** [comment]
另外請問一下，這些詞庫中與八股文重複的詞，我應該是要拿掉呢？還是留著無妨？

**Reply by lotem** [reply]
不需要拿掉。

**Reply by markwu** [reply]
了解！

**Comment by LEOYoon-Tsaw** [comment]
如果兩個詞庫有交集，可以用交集部分訓練一個隨機森林模型，例如，基於新詞庫詞頻預測八股文詞頻，然後使用預測詞頻替換新詞庫中的原有詞頻，應可大體保證最佳體驗

**Reply by markwu** [reply]
謝謝，這已經超出我知識範圍了，請問有可以相關的參考網頁或作法嗎？

**Comment by lotem** [comment]
怎樣比較好，取決於來源詞庫的數據質量。
不開八股文，大體能保持來源詞庫的排序。
開八股文，兩個來源的詞頻會累加起來。

**Reply by markwu** [reply]
了解，請問兩個不同表示方法式的詞頻和八股文會如何累加呢？

例如八股文是用次數，而酷音輸入法的詞庫是用百分比，我需要事先做轉換嗎？

**Reply by lotem** [reply]
酷音的百分比基數是什麼？

Rime 詞典會用百分數乘八股文的詞頻。

**Reply by markwu** [reply]
@lotem 謝謝，我參考其他的轉檔程式把百分比改為次數了。
